app/controllers/comments_controller.rb

```diff
@@ -40,54 +35,30 @@ def edit
   # POST /comments
   # POST /comments.json
   def create
-    params[:comment][:author_id] = current_member.id
     @comment = Comment.new(comment_params)
-
-    respond_to do |format|
-      if @comment.save
-        format.html { redirect_to @comment.post, notice: "Comment was successfully created." }
-        format.json { render json: @comment, status: :created, location: @comment }
-      else
-        format.html { render action: "new" }
-        format.json { render json: @comment.errors, status: :unprocessable_entity }
-      end
-    end
+    @comment.author = current_member
+    flash[:notice] = "Comment was successfully created." if @comment.save
+    respond_with(@comment.post)
   end
 
   # PUT /comments/1
   # PUT /comments/1.json
   def update
-    # you should never be able to change the author or post when
-    # updating
-    params[:comment].delete("post_id")
-    params[:comment].delete("author_id")
-
-    respond_to do |format|
-      if @comment.update(comment_params)
-        format.html { redirect_to @comment.post, notice: 'Comment was successfully updated.' }
-        format.json { head :no_content }
-      else
-        format.html { render action: "edit" }
-        format.json { render json: @comment.errors, status: :unprocessable_entity }
-      end
-    end
+    flash[:notice] = 'Comment was successfully updated.' if @comment.update(comment_params)
+    respond_with(@comment.post)
   end
 
   # DELETE /comments/1
   # DELETE /comments/1.json
   def destroy
     @post = @comment.post
     @comment.destroy
-
-    respond_to do |format|
-      format.html { redirect_to @post }
-      format.json { head :no_content }
-    end
+    respond_with(@post)
   end
 
   private
 
   def comment_params
-    params.require(:comment).permit(:author_id, :body, :post_id)
+    params.require(:comment).permit(:body, :post_id)
```

Comment:
note: you can't set the author_id ever.
the old code was allowing the params, and the deleting it (see line 62 of old half of diff)